iree/turbine/kernel/compiler/kernel_codegen.py

```diff
@@ -251,13 +251,34 @@ def determine_input_output_buffers(self, graph: fx.Graph):
         # Extract all placeholder nodes.
         placeholder_nodes = filter_fx_graph(graph, is_placeholder)
 
+        def get_users_recursive(node):
```

Comment:
@GMNGeoffrey - this could explain why you had to mess with the input/output bindings. (We didn't look recursively to identify output nodes)

Reply:
Maybe. I think the logic is still wrong though as noted in https://github.com/iree-org/iree-turbine/issues/375. If a node has both read and write dependencies it can't be marked as read-only, which is still what happens here IIUC.